File: Android project/SharedPreferenceArrayListGson/app/src/main/java/com/mirkamol/sharedpreferencearraylistgson/adapter/UserAdapter.java
package com.mirkamol.sharedpreferencearraylistgson.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.mirkamol.sharedpreferencearraylistgson.R;
import com.mirkamol.sharedpreferencearraylistgson.model.User;

import java.util.ArrayList;

public class UserAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
    ArrayList<User> items = new ArrayList<>();


    public UserAdapter(ArrayList<User> items) {
        this.items = items;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_layout, parent, false);
        return new UserViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
      if (holder instanceof UserViewHolder){
          ((UserViewHolder) holder).fullName.setText(items.get(position).getFullName());
          ((UserViewHolder) holder).email.setText(items.get(position).getEmail());
      }

    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    static class UserViewHolder extends RecyclerView.ViewHolder{

        TextView fullName;
        TextView email;
        public UserViewHolder(@NonNull View itemView) {
            super(itemView);


            fullName = itemView.findViewById(R.id.full_name);
            email = itemView.findViewById(R.id.tv_email);
        }
    }
}

File: Android project/FirebaseExampleSanjarSuvonov/app/src/main/java/com/mirkamol/firebaseexamplesanjarsuvonov/adapter/UserAdapter.kt
package com.mirkamol.firebaseexamplesanjarsuvonov.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.mirkamol.firebaseexamplesanjarsuvonov.R
import com.mirkamol.firebaseexamplesanjarsuvonov.model.User

class UserAdapter(var users: ArrayList<User>) : RecyclerView.Adapter<UserAdapter.Vh>() {
    class Vh(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun onBind(user: User) {
            val tv = itemView.findViewById<TextView>(R.id.tv)
            tv.text = user.name + user.email + user.password
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Vh {
        return Vh(LayoutInflater.from(parent.context).inflate(R.layout.item_layout, parent, false))
    }

    override fun onBindViewHolder(holder: Vh, position: Int) {
        holder.onBind(users[position])
    }

    override fun getItemCount(): Int = users.size
}